extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/DefaultTemplateExtensions.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.qute.runtime;
+
+import static io.quarkus.qute.TemplateExtension.ANY;
+
+import java.util.Map;
+
+import io.quarkus.qute.Results.Result;
+import io.quarkus.qute.TemplateExtension;
+
+public class DefaultTemplateExtensions {
+
+    @SuppressWarnings("rawtypes")
+    @TemplateExtension(matchName = ANY)
+    public static Object map(Map map, String name) {
```

Comment:
So what's the rationale for deciding to put this in the extension rather than in Qute-core?

Reply:
I don't remember ;-). I need to check the history.

Reply:
Haha.

Reply:
Hm, I think I remember now - the problem is that programatically added resolvers are ignored during validation of param declarations. So if there was `map.foo` it would fail even if there is a map entry for key `foo`. This area will definitely need more tests and improvements.

Reply:
Ah, I guess probably the programmatic API doesn't have as much typing info as the annotation way? I thought the annoation would strictly use the programmatic API.

Reply:
Well, most of the core APIs currently don't care about "type-safe validation" at all. It's an optional feature implemented on the quarkus integration level. Of course, we could try to change it later.

Reply:
Ah, OK, I get it. In that case I suggest to move the validation up into Qute, even if you make it default to non-validated outside of Quarkus.

OR, perhaps a simpler trick would be to add just the typing info to the programmatic API, and make the Quarkus extension use this for checking?